File: public/dsa/arrays/identical_twins/hashing_approach_1/identical_twins.cpp

#include <bits/stdc++.h>
using namespace std;

int getIdenticalTwinsCount(vector<int> &arr) {
    int count = 0;
    unordered_map<int, int> frequency;
    for (auto v : arr) {
        count += frequency[v];
        frequency[v]++;
    }
    return count;
}
// tests-start
int main() {
    vector<int> arr = {1, 2, 3, 2, 1};
    int count = getIdenticalTwinsCount(arr);
    cout << count << endl;
    return 0;
}
// tests-end

/*output
2
*/

File: public/dsa/arrays/identical_twins/hashing_approach_1/identical_twins.py
from collections import defaultdict
from typing import List


class Solution:
    def getIdenticalTwinsCount(self, arr: List[int]) -> int:
        count = 0
        mp = defaultdict(int)
        for v in arr:
            count += mp[v]
            mp[v] += 1
        return count


# tests-start
arr = [1, 2, 3, 2, 1]
sol = Solution()
print(sol.getIdenticalTwinsCount(arr))
# tests-end

'''output
2
'''
